### PR TITLE
Fix #10973. AnalyticsUserCountsWithPostsByDay is sync now

### LINKED ISSUE #10973: Migrate "Post.AnalyticsUserCountsWithPostsByDay" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15855)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `AnalyticsUserCountsWithPostsByDay` in the `Post` store.

The expected way to implement it is, go to the `Post` store implementation in the `store/sqlstore/` directory, modify the method `AnalyticsUserCountsWithPostsByDay` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### LINKED ISSUE #10975: Migrate "Post.GetEtag" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15853)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetEtag` in the `Post` store.

The expected way to implement it is, go to the `Post` store implementation in the `store/sqlstore/` directory, modify the method `GetEtag` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by rvillablanca**
Should resolve  #10973

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -70,20 +70,22 @@ require (
 	github.com/spf13/cobra v0.0.3
```
Comment (jespino) [inline go.mod]: You shouldn't be modifying go.mod or go.sum

File: store/sqlstore/post_store.go [hunk 2]
```diff
@@ -958,57 +958,55 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
...
-		}
-	})
+	var rows model.AnalyticsRows
+	_, err := s.GetReplica().Select(
+		&rows,
+		query,
+		map[string]interface{}{"TeamId": teamId, "StartTime": start, "EndTime": end})
+	if err != nil {
+		return nil, model.NewAppError("SqlPostStore.AnalyticsUserCountsWithPostsByDay", "store.sql_post.analytics_user_counts_posts_by_day.app_error", nil, err.Error(), http.StatusInternalServerError)
+	} else {
```
Comment (jespino) [inline store/sqlstore/post_store.go]: You can remove the else and just return at the top level of the function

File: store/sqlstore/post_store.go [hunk 3]
```diff
@@ -958,57 +958,55 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
...
 }
 
-func (s *SqlPostStore) AnalyticsUserCountsWithPostsByDay(teamId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		query :=
-			`SELECT DISTINCT
+func (s *SqlPostStore) AnalyticsUserCountsWithPostsByDay(teamId string) (model.AnalyticsRows, *model.AppError) {
+	query :=
+		`SELECT DISTINCT
 			        DATE(FROM_UNIXTIME(Posts.CreateAt / 1000)) AS Name,
```
Comment (jespino) [inline store/sqlstore/post_store.go]: You should be maintaining the query identation consistent, here and in the rest of the changes of this function.

**Comment by rvillablanca** [comment]
Finished all requested changes @jespino

**Comment by jespino** [comment]
@rvillablanca do you mind to merge master when you can?

**Comment by rvillablanca** [comment]
Ok @jespino I just did it ;)

**Comment by jespino** [comment]
Merged! Thanks @rvillablanca! :tada:

**Pull request merged**